### DISCUSSION TITLE
e2e type safety

### DISCUSSION THREAD

**Discussion by matthewlilley** [wevm/wagmi#682] [Ideas] [answered]
Curious about thoughts on achieving e2e type safety between blockchain -> wagmi/core -> wagmi, since the server (blockchain) already has a type definitions (abi), this should be doable and would avoid seeing things like this on writeAsync, etc...

![Screenshot_325](https://user-images.githubusercontent.com/6063350/177889292-c4956e12-521c-4925-a36c-b32767d788a4.png)
 
Prisma, tRPC and others are able to achieve this with the use of zod.

https://github.com/colinhacks/zod

**Comment by tmm** [comment, marked as answer]
Coming to wagmi soon in https://github.com/wagmi-dev/wagmi/pull/941! Powered by the [ABIType](https://github.com/wagmi-dev/abitype).
